qualtran/bloqs/arithmetic/hamming_weight.py

```diff
@@ -63,6 +64,21 @@ def signature(self):
             ]
         )
 
+    @cached_property
+    def junk_bitsize(self) -> SymbolicInt:
+        return self.bitsize - self.bit_count_of_bitsize
+
+    @cached_property
+    def out_bitsize(self) -> SymbolicInt:
+        return bit_length(self.bitsize)
+
+    @cached_property
+    def bit_count_of_bitsize(self) -> SymbolicInt:
+        """lower bound on number of 1s in bitsize"""
+        if is_symbolic(self.bitsize):
+            return 1  # worst case
```

Comment:
Can you open an issue, and link it here, with a request that we should add support for custom sympy functions so we don't need to assume a worst case bound here? Ideally, we would define a `bit_count(x)` function in `symbolics/` and report the costs in terms of `bit_count(N)`. When users have values of `N`, one could substitute it in the symbolic cost expression and `bit_count(N)` would evaluate to a number as expected.